### PR TITLE
38-ts-countdown-timer

### PR DESCRIPTION
**Pull request by Aanndreas**
This code sets up a countdown timer functionality for communication between the server and clients.

In app.js:

When the server emits a 'startGame' event, it triggers a countdown timer that starts at 60 seconds.
The countdown is updated every second, and if it reaches zero, the 'countdownFinished' event is emitted to notify clients that the countdown has ended.
In main.ts:

There is a function "updateCountdownDisplay" that updates the countdown display on the client side.
An event listener is added to the startGameButton (which is currently hidden) so that when it's clicked, a 'startGame' event is emitted to the server.
Event listeners are set up to listen for 'countdownUpdate' and 'countdownFinished' events from the server. When a 'countdownUpdate' event is received, the updateCountdownDisplay function is called to update the countdown display. When a 'countdownFinished' event is received, it displays a message indicating that the time is up.
Additionally, there are temporary code blocks for testing purposes:

A test button (testStartCountdownButton) is added to trigger the countdown. This is meant for development purposes and should be removed in production.
An event listener is added to this test button to emit a 'startGame' event to the server when clicked.
The HTML structure contains the elements necessary for displaying the game UI, including the countdown timer (#countdownValue), login section, game lobby section, and other game-related sections.

### CODE REVIEW

File: client/assets/images/gamePage.png
Comment (RalfiSlask) [inline client/assets/images/gamePage.png:1]: Orimligt snygg! 😁

File: client/index.html [hunk 1]
```diff
@@ -12,14 +12,17 @@
...
+        <div id="countdownMessage"></div>
       </div>
     </header>
     <main>
       <section id="loginSection">
         <label for="loginInput">Username:</label>
         <input id="loginInput" type="text" />
         <button id="loginButton">Login</button>
+        <!-- Test button to trigger countdown (temporary, remove in production) -->
+        <button id="testStartCountdownButton">Test Start Countdown</button>
```
Comment (RalfiSlask) [inline client/index.html:25]: Bra att implementera placeholder och där ha kommentar för oss andra 👍

File: server/app.js [hunk 2]
```diff
@@ -66,6 +66,24 @@ io.on('connection', socket => {
     users.splice(userIndex, 1);
     io.emit('updateUserList', users);
   });
+
+// Listen for the "startGame" event from the server
```
Comment (RalfiSlask) [inline server/app.js:70]: Denna kommentar är lite överflödig, men sådant vi kan rensa i slutet 🙂

File: server/app.js [hunk 3]
```diff
@@ -66,6 +66,24 @@ io.on('connection', socket => {
...
+socket.on('startGame', () => {
+  let countdown = 60; // Initial countdown value in seconds
+
+  // An interval to update the countdown every second
+  const countdownInterval = setInterval(() => {
+    if (countdown <= 0) {
+      // If countdown reaches zero, clear the interval and emit a "countdownFinished" event
```
Comment (RalfiSlask) [inline server/app.js:77]: Good Job maaan 💯

**Review by theufoparty — APPROVED** [review]

**Pull request merged**